### PR TITLE
Store DB to shared container on iOS

### PR DESCRIPTION
**Pull request by Nils-Becker**

### CODE REVIEW

**Review by kaiwidmer — APPROVED** [review]
👍

File: shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
 import com.squareup.sqldelight.db.SqlDriver
 import com.squareup.sqldelight.drivers.native.NativeSqliteDriver
+import com.squareup.sqldelight.drivers.native.wrapConnection
+import platform.Foundation.NSFileManager
 
 actual class DriverFactory : DriverCreator {
     actual override fun createDriver(): SqlDriver {
-        return NativeSqliteDriver(Database.Schema, "app.db")
+        val groupUrl = NSFileManager.defaultManager
+            .containerURLForSecurityApplicationGroupIdentifier("group.ch.dreipol.reZHycle")
```
Comment (tschuls) [inline shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt:13]: Can we move this to a constant?

File: shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+        val schema = Database.Schema
+        val config = DatabaseConfiguration(name = "app.db",
```
Comment (tschuls) [inline shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt:15]: Can we move this to a constant?

File: shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+            basePath = groupUrl!!.path,
```
Comment (tschuls) [inline shared/src/iosMain/kotlin/ch/dreipol/multiplatform/reduxsample/shared/database/DriverFactory.kt:16]: I assume you want the app to crash, if groupURL is null, correct? I would prefer a Excpetion with a message, to not have a forced unwrap.

**Review by tschuls — CHANGES_REQUESTED** [review]

**Review by tschuls — APPROVED** [review]
Merci

**Pull request merged**